webkit2/tests/tests.lisp

```diff
@@ -0,0 +1,258 @@
+;;; tests/tests.lisp -- Testing code.
+
+;; This file is part of cl-webkit.
+;;
+;; cl-webkit is free software; you can redistribute it and/or modify
+;; it under the terms of the MIT license.
+;; See `COPYING' in the source distribution for details.
+
+;;; Code:
+
+(in-package :webkit2/tests)
+
+(def-suite webkit-tests :description "Testing WebKit.")
+
+(def-suite js-tests :description "Testing JS value transformation." :in webkit-tests)
+
+(defvar *webkit-environment* (make-hash-table :test 'equal))
+
+(defmacro with-js-transform-result ((var js-string) &body body)
+  `(let ((channel (make-instance
+                   'calispel:channel
+                   :buffer (make-instance 'jpl-queues:bounded-fifo-queue :capacity 1))))
+     (gtk:within-gtk-thread
+       (webkit2:webkit-web-view-evaluate-javascript
+        (gethash "view" *webkit-environment*) ,js-string
+        (lambda (result) (calispel:! channel result))))
+     (let ((,var (calispel:? channel)))
+       ,@body)))
+
+(gtk:within-main-loop
```

Comment:
This should belong to a setup function, no?  Otherwise it would start the GTK loop when asdf:load-ing.